### DISCUSSION TITLE
Questions of DICE Cert/Key generation model

### DISCUSSION THREAD

**Discussion by yh36** [chipsalliance/caliptra-sw#1642] [Q&A] [answered]
1. In the "figure 4: DICE Cert/Key generation" of [Main Caliptra specification](https://github.com/chipsalliance/Caliptra/blob/main/doc/Caliptra.md), how does this Identity model comply with both [DICE hardware requirements](https://trustedcomputinggroup.org/wp-content/uploads/Hardware-Requirements-for-a-Device-Identifier-Composition-Engine-Version-1.0-Revision-0.91_pub.pdf) and [DICE Layering Architecture](https://trustedcomputinggroup.org/wp-content/uploads/DICE-Layering-Architecture-r19_pub.pdf)? The basic idea in DICE hardware requirements is for DICE to derive the device identity, i.e., CDI, using both UDS and TCI<sub>FMC</sub>. So, in the Caliptra Identity model, I think the "Caliptra FMC" is treated as the DICE's FMC (i.e., Layer 0) and "Caliptra ROM" becomes the DICE Layer. Then the CDI<sub>FMC</sub> in the figure 4 does satisfy the DICE requirements. But the provision of IDEVID seems not to follow the approaches as described in the spec of DICE layering architecture. In the section 9.2.3.1 and 9.2.3.2 of the layering spec, both approaches generate CSR of IDEVID from Layer 0. But in Caliptra, the CSR of IDEVID comes from DICE layer and this IDEVID only contains UDS info without using the TCI info from its FMC. Does this mean that Caliptra's identity model does not strictly follow TCG DICE layering structure?
2. The whole DICE Cert/Key generation (including DPE) only supports ECC asymmetric key. I can see images in secure boot have PQC support. How about attestation? Do we support attestation using PQC algorithm?

**Comment by jhand2** [comment, marked as answer]
1. Here are a couple relevant sections from the layering spec:
    * [6.2.1] "At layers above the DICE HRoT layer, the CDI value received from the previous TCB component supplies a statistically unique value to the current TCB component."
    * [6.2.1] "The TCI value for any TCB component that includes firmware or software MUST include measurement of said firmware or software"

However, Caliptra DICE is rooted in ROM, which is considered the "Hardware" layer from this spec. These sections make explicit exceptions for Hardware.

The examples you cited in section 9 are examples of rooting DICE in Layer 0. But they are just examples (see figure 11 which is explicitly marked as an example). It is valid to instead root DICE in Hardware and use the hardware-layer's CDI to derive IDevID, which does not contain firmware measurements.

2. We will support PQC attestation in 2.0. Not in 1.0.
